### PR TITLE
Add Proxy Submission Acceptance Tests

### PR DESCRIPTION
**Pull request by frogfoot9013**
Add acceptance tests for proxy submissions on behalf of submitter with and without a PASS account, rework the previous tests to split across multiple files for convenience.

### CODE REVIEW

File: tests/commonTest.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    await t
+        .click('#login-button')
+        .typeText(Selector('#username'), 'admin-submitter')
+        .typeText(Selector('#password'), 'moo')
+        .click(Selector(".form-element.form-button"));
+});
+
+const currLocation = ClientFunction((() => window.location.href));
+
+module.exports = {
```
Comment (jabrah) [inline tests/commonTest.js:37]: We should keep consistent module formats, so prefer ESM syntax. For example here, the `module.exports` should be removed in favor of `export const userIncompleteNih = ...`. In this example, we'd use `import` like `import { userIncompleteNih } from './commonTests';`

File: tests/proxySubmissionTests.js [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+    await t.click(userHasGrantsLink);
+    await t.expect(searchResultsModal.exists).notOk();
+
+    const selectedUser = Selector('p').withText('Currently selected submitter:');
+    await t.expect(selectedUser.innerText).contains("Staff Hasgrants");
+    
+    await walkThroughSubmissionFlow(t, true);
+
+}).timeouts({
+    pageLoadTimeout:    60000,
```
Comment (jabrah) [inline tests/proxySubmissionTests.js:57]: Would be good to have timeouts configurable with environment variables, so we could change them in CI, for example, without a code change

**Review by jabrah — CHANGES_REQUESTED** [review]
Looking pretty good overall. Updating the timeouts to use environment variables can be a future update, but I think we should update the module format

File: tests/commonTest.js (hunk 1 above)
Comment (frogfoot9013) [reply tests/commonTest.js]: Fixed this.

File: tests/proxySubmissionTests.js (hunk 2 above)
Comment (frogfoot9013) [reply tests/proxySubmissionTests.js]: Not sure if I've done this the right way, I think I might need to have a different way of doing it.

**Review by jaredgalanis — APPROVED** [review]
Looks good to me 👍

**Comment by jaredgalanis** [comment]
> Looks good to me 👍

So this approval was just based on a review of the code. If things aren't running well yet, I'll wait until you get there.

File: tests/commonTest.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+        .click('#login-button')
+        .typeText(Selector('#username'), 'admin-submitter')
+        .typeText(Selector('#password'), 'moo')
+        .click(Selector(".form-element.form-button"));
+});
+
+export const currLocation = ClientFunction((() => window.location.href));
+
+// set in milliseconds
+export const TIMEOUT_LENGTH = parseInt(process.env.TIMEOUT_LENGTH) || 60000;
```
Comment (jabrah) [inline tests/commonTest.js:38]: I think for now, this is a fine way of reading in an environment variable.

If you need to add more environment variables down the line (such as for other Testcafe timeouts), then you might consider using a `.env` file with an NPM package like `dotenv`, but for that's more complexity than we need right now.

**Review by jabrah — COMMENTED** [review]
Ran tests with the command `TIMEOUT_LENGTH=30000 npx testcafe chrome tests/` successfully. Manually checking the timeout value in the tests showed that the value did respond to the environment variable fine.

One request: could you update the README to include some simple testing instructions, including a description of the environment variable(s) available

File: tests/baseCaseTests.js [hunk 4]
```diff
@@ -242,8 +209,4 @@ test('can walk through an nih submission workflow and make a submission - base c
     const submittedFile = Selector('a').withExactText('my-submission.pdf');
     await t.expect(submittedFile.exists).ok();
 
-}).timeouts({
-    pageLoadTimeout:    60000,
-    pageRequestTimeout: 60000,
-    ajaxRequestTimeout: 60000,
-});;
+}).disablePageCaching;
```
[the remark this answers is not in the document]
Comment (frogfoot9013) [reply tests/baseCaseTests.js:212]: I don't actually think those removed timeouts seemed to do anything, at least in the context of what I was doing. I think they're for waiting for a page to load, as opposed to what I have been working with timeouts, which has been for getting a selector to wait for a particular element to appear (which is set when creating the Selector itself).

**Review by jabrah — APPROVED** [review]

**Pull request merged**